src/test/SerializeTool_test.cpp

```diff
@@ -0,0 +1,593 @@
+/*
+    This file is part of ripple-sign-serialize:
+        https://github.com/ximinez/ripple-sign-serialize
+    Copyright (c) 2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <SerializeTool.h>
+#include <RippleKey.h>
+#include <Serialize.h>
+#include <test/KnownTestData.h>
+#include <test/KeyFileGuard.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/SecretKey.h>
+#include <boost/format.hpp>
+
+namespace serialize {
+
+namespace test {
+
+class SerializeTool_test : public beast::unit_test::suite
+{
+private:
+
+    // Allow cout to be redirected.  Destructor restores old cout streambuf.
+    class CoutRedirect
+    {
+    public:
+        CoutRedirect(std::stringstream& sStream)
+            : old_(std::cout.rdbuf(sStream.rdbuf()))
+        {
+        }
+
+        ~CoutRedirect()
+        {
+            std::cout.rdbuf(old_);
+        }
+
+    private:
+        std::streambuf* const old_;
+    };
+
+    class CInRedirect
+    {
+    public:
+        CInRedirect(std::stringstream& sStream)
+            : old_(std::cin.rdbuf(sStream.rdbuf()))
+        {
+        }
+
+        ~CInRedirect()
+        {
+            std::cin.rdbuf(old_);
+        }
+
+    private:
+        std::streambuf* const old_;
+    };
+
+    void
+    testSerialize()
+    {
+        testcase("Serialize");
+
+        auto test = [&](TestItem const& testItem)
+        {
+            {
+                std::stringstream capture;
+                CoutRedirect coutRedirect{ capture };
+
+                doSerialize(testItem.JsonText);
+
+                BEAST_EXPECT(capture.str() == testItem.SerializedText + "\n");
+            }
+            {
+                std::stringstream jsoninput(testItem.JsonText);
+                CInRedirect cinRedirect{ jsoninput };
+
+                std::stringstream capture;
+                CoutRedirect coutRedirect{ capture };
+
+                runCommand("serialize", {}, {});
+
+                BEAST_EXPECT(capture.str() == testItem.SerializedText + "\n");
+            }
+        };
+
+        test(getKnownTx());
+        test(getKnownMetadata());
+        //test(getKnownLedger());
```

Comment:
As above, I was hoping to support it, but it makes more sense to re-add it later.